src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -240,7 +241,14 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuin
 
             // Read the first vector unaligned.
 
-            currentMask = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
+            if (Sse2.IsSupported)
+            {
+                currentMask = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
+            }
+            else
+            {
+                currentMask = AdvSimd.Arm64.MaxAcross(AdvSimd.LoadVector128(pBuffer)).ToScalar();
```

Comment:
>currentMask = AdvSimd.Arm64.MaxAcross(AdvSimd.LoadVector128(pBuffer)).ToScalar(); [](start = 16, length = 81)

Is `MaxAcross` correct? From what I understand, lets say that you have following vector:
`<0, 0, 240, 0, 126, 0, 0, 151, 0, 0, 0, 0, 0, 0, 0, 0>`. 
`Sse2.MoveMask()` would return `132 (0000 0000 1000 0100)` and then trailing zero count would be 2.

But in your case, `MaxAcross` would return `240` and trailing zero count would return 4.

You might to do something like this:
[https://github.com/dotnet/runtime/blob/0d5e5766c6688a5ab29d6e96c01abbfe1ad3659c/src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs#L1943](https://github.com/dotnet/runtime/blob/0d5e5766c6688a5ab29d6e96c01abbfe1ad3659c/src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs#L1943)